amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpRpcFlowStage.scala

```diff
@@ -174,18 +168,37 @@ final class AmqpRpcFlowStage(settings: AmqpSinkSettings, bufferSize: Int, respon
             if (queue.nonEmpty) {
               pushMessage(queue.dequeue())
             }
+
+          override def onDownstreamFinish(): Unit = {
+            setKeepGoing(true)
+            if (unackedMessages.get() == 0) super.onDownstreamFinish()
```

Comment:
This means that if a user forgot to ack/nack the stream will never complete, imagine doing a filter or having a buffer somewhere that has unconsumed elements in it. 

I'm not sure how to deal with this in a good way, I think a timeout of some kind may be the only solution and then either fail the stage or nack all those elements after the timeout. Wdyt?

Reply:
This is the way akka streams kafka does it. And I think that is a good way (I had exactly this problem last week).

If the user forgets to ack/nack (I did forget to nack bogus messages), he'll soon reach the inflight limit and the queue will block making it very obvious that messages are not flowing and not wasting messages or modifying the order of the queue.

Setting a timeout will essentially hide the issue.
Also, there are few thing:
* I'm not sure if nacking will preserve the order or if it's better to just fail and disconnect the client.
* How do we set the timeout? based on the oldest unacked message? Then we would need to store the time with the messages and store the list in the queue. Based on the latest message? This might be easier
* I think that the timeout will need to be configurable since we don't know how time-consuming the processing that the users are doing is.
* Is there a possibility that, if you are using some retry supervision like backoff, you might end up in a loop where everything seems to be running but nothing is actually happening??

Wdyt?

Reply:
There is a big difference with Kafka though, where there isn't really individual message-acks but  rather a highest-seen-committed counter. But maybe keep it as is and see if people have problems with it and solve that if so.

If we implement it, I think it should you are right it should be just failing the stage. The timeout could be from configuration, with a generous default, the timer would start when downstream or upstream completing, if the timeout hits and there still are outstanding non-acked (in counter) then we fail the stage and tear down the connection (I'm guessing the server will see that as nack:s then for the outstanding messages).

Reply:
I think that it's a good idea to keep it as it is and act if people find problems.
If you are happy with that I'll consider this done.

If you still consider the timeout necessary, just let me know :)

Reply:
Let's merge it like this and see if is surprising to people using it.